### PR TITLE
Have Writer's Return a Response

### LINKED ISSUE #820: WhyLabsWriter write() returns None
## Description

The WhyLabsWriter write() used to return the response from the API call, but that changed on the latest updates. Now it returns None.

The method should let the user know whether the write went through.

- System info
  - Operating system and version: 
  - Browser and version: 
- Steps to reproduce:
  - Step
  - Next step

## Related

Relates to organization/repo#number 

- [x] I have reviewed the [Guidelines for Contributing](https://github.com/whylabs/whylogs/blob/mainline/.github/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/whylabs/whylogs/blob/mainline/.github/CODE_OF_CONDUCT.md).

### PR DESCRIPTION
**Pull request by TheMellyBee**
## Description
WhyLabsWriter returned the response in v0 but currently returns None. Changed the return for write to be a Response or a Boolean as many of the other writers do not return a response., but thought it would be nice for the same reason. 

## Related

Closes  #820

- [ X] I have reviewed the [Guidelines for Contributing](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md).

### CODE REVIEW

File: python/whylogs/api/writer/writer.py [hunk 1]
```diff
@@ -20,7 +22,7 @@ def get_default_path(self) -> str:
         pass
 
     @abstractmethod
-    def write(self, path: Optional[str] = None, **kwargs: Any) -> None:
+    def write(self, path: Optional[str] = None, **kwargs: Any) -> requests.Response:
```
Comment (jamie256) [inline python/whylogs/api/writer/writer.py]: Two issues:
1. We can't reference an optional dependency's type ('requests' package is only used in whylogs[whylabs]) in our base abstract classes used across all Writers/Writables
2. I would expect the write() method would have the same return type across Writer and Writable

File: python/whylogs/api/writer/writer.py [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import os
 from abc import ABC, abstractmethod
-from typing import Any, Optional, TypeVar
+from typing import Any, Optional, TypeVar, Union
+
+import requests  # type: ignore
```
Comment (jamie256) [inline python/whylogs/api/writer/writer.py]: we can't import this package as a top level import in Writers, it is not a dependency of whylogs and shouldn't be. 'requests' should be specific to the [whylabs] extra and any reference needs to guard against cases where users don't have it (see PIL or pandas import patterns)

**Review by jamie256 — CHANGES_REQUESTED** [review]
Also check the CI failures.

File: python/whylogs/api/writer/writer.py (hunk 2 above)
Comment (TheMellyBee) [reply python/whylogs/api/writer/writer.py]: Got it. So only in the whylabs one then. :) I can change this to an any type for the abstract return which will be able to keep this from being here

File: python/whylogs/api/writer/writer.py (hunk 1 above)
Comment (TheMellyBee) [reply python/whylogs/api/writer/writer.py]: 1. Got it. 
2. That's fair, that makes sense. Hmm ok, well then. One option is that we return Any for all writables. That's a little odd but also kinda understandable. We could return an int for all writes which would be more understandable with the error codes. 

Or a wrapper around the write just for whylabs. I don't like this. 

Or lastly, not add the response which would force use of the private method _do_upload instead. Also do not like. 

My favorite would be the return any or int. Thoughts?

Comment (jamie256) [reply python/whylogs/api/writer/writer.py]: `Any` works, or maybe something like `Tuple[bool, str]` if this is just to give success/fail status + some debug / status information. The string part could be: filepath, url, in success case, or error or http status code in the failure case?

Comment (TheMellyBee) [reply python/whylogs/api/writer/writer.py]: Didn't see this before latest commit. I had moved it to an Optional[int] where most don't return anything. I can iterate and move it to that. I thought the direct response of the status code would be nice, but I agree that has more information to give

**Review by jamie256 — APPROVED** [review]
Looks good! just need to resolve your conflicts

**Pull request merged**